### ISSUE TITLE
Peer review of the OpenBioLinks manuscript for Bioinformatics

### ISSUE THREAD

**Issue by dhimmel** [OpenBioLink/OpenBioLink#3]
I'm reviewing the following OpenBioLinks manuscript for the journal _Bioinformatics_:

> **OpenBioLink: A resource and benchmarking framework for large-scale biomedical link prediction**  
Anna Breit, Asan Agibetov, Matthias Samwald  
*arXiv* (2019-12-10) <https://arxiv.org/abs/1912.04616v1>

I'm going to create GitHub issues for my major pieces of feedback. The following list contains the issues that will compose my review.

- [x] post the study as a preprint https://github.com/OpenBioLink/OpenBioLink/issues/1
- [x] package the python code https://github.com/OpenBioLink/OpenBioLink/issues/2
- [x] improve data documentation and add formats https://github.com/OpenBioLink/OpenBioLink/issues/4
- [x] consider copyright and license restrictions of upstream data sources https://github.com/OpenBioLink/OpenBioLink/issues/5
- [x] apply an open source software license https://github.com/OpenBioLink/OpenBioLink/issues/6
- [x] questions regarding the train and testing splitting https://github.com/OpenBioLink/OpenBioLink/issues/7
- [x] does it make sense to combine assessment of edge prediction across all edge types https://github.com/OpenBioLink/OpenBioLink/issues/8

## Summary

The study proposes OpenBioLink, which is a network as well as a benchmark dataset for assess hetnet edge prediction techniques. A standardized benchmark to evaluate hetnet edge prediction is of major importance to the field, and I'm not aware of existing resources for this task.

However, several issues must be addressed for OpenBioLink to be adopted as a community benchmarking dataset. My review focuses on these issues, which I've posted as a series of GitHub Issues at https://github.com/OpenBioLink/OpenBioLink/issues/3.
 
The manuscript currently focuses on embedding-based methods for edge predictions on hetnets. That is okay with me, but the manuscript currently reads as if there are not preexisting non-embedding methods. For example, my research in [Project Rephetio](https://doi.org/10.7554/eLife.26726) applied a metapath-based approach to predict edges in a biomedical hetnet named Hetionet.

In conclusion, I think a the benchmarking dataset could be of great value. However, this depends on the authors ability to either recruit many approaches to submit predictions. I look forward to the authors addressing my suggestions on how to make this feasible.

**Comment by matthias-samwald** [comment]
Thanks for the very detailed review!

**Comment by matthias-samwald** [comment]
I have now submitted a revised version of the manuscript (still being processed).

>>>
The study proposes OpenBioLink, which is a network as well as a benchmark dataset for assess hetnet edge prediction techniques. A standardized benchmark to evaluate hetnet edge prediction is of major importance to the field, and I'm not aware of existing resources for this task.

However, several issues must be addressed for OpenBioLink to be adopted as a community benchmarking dataset. My review focuses on these issues, which I've posted as a series of GitHub Issues at https://github.com/OpenBioLink/OpenBioLink/issues/3.

- [x] post the study as a preprint https://github.com/OpenBioLink/OpenBioLink/issues/1
<<<

→ The study has been published as a preprint of ArXiv: https://arxiv.org/abs/1912.04616 (also linked from GitHub page)



>>>
- [ ] package the python code https://github.com/OpenBioLink/OpenBioLink/issues/2 
<<<

→ The code has been packaged and can now be installed through pip.



>>>
- [ ] improve data documentation and add formats https://github.com/OpenBioLink/OpenBioLink/issues/4 
<<<

→ We have updated the data documentation and added RDF as an additional data format that can be readily important into a wide variety of tools and graph database systems. The choice of RDF also eases the integration of OpenBioLink resources with other Linked Data resources that we and some of our collaborators have created or worked with in the past.




>>> 
- [ ] consider copyright and license restrictions of upstream data sources https://github.com/OpenBioLink/OpenBioLink/issues/5 
<<<

→ We have adopted the approach taken by HetioNet, i.e., licensing any original contributions to the dataset by our project as CC0 while maintaining the licensing restrictions of data items from external datasets. We have extended the GitHub ReadMe with a table detailing the licensing terms of integrated datasets and and a reminder that they should be taken into consideration when redistributing the benchmark or any derivatives. 
We have added information on the source of each data item (i.e., relation) to the benchmark files. 
We added the following sentence to the manuscript: “Licensing terms of integrated datasets are detailed in the documentation, and should be taken into consideration when redistributing the benchmark or any derivative work.”


>>> 
- [ ] apply an open source software license https://github.com/OpenBioLink/OpenBioLink/issues/6 
<<<

→ The code has now been released under the MIT license.



>>> 
- [ ] questions regarding the train and testing splitting <https://github.com/OpenBioLink/OpenBioLink/issues/7 
<<<

→ We have added information on which data sources currently allow for time-sliced benchmark creation to the table listing data sources in the README. We will investigate the effect of degree-preserving graph permutation (such as those produced by XSwap) and alternative sampling technques such as Bernoulli negative sampling in follow-up work.



>>> 
- [ ] does it make sense to combine assessment of edge prediction across all edge types https://github.com/OpenBioLink/OpenBioLink/issues/8 
<<<

→  Per-relation evaluation results have been added to the supplementary material (Table S5), the manuscript has been updated to refer to the supplementary material.



>>> 
The manuscript currently focuses on embedding-based methods for edge predictions on hetnets. That is okay with me, but the manuscript currently reads as if there are not preexisting non-embedding methods. For example, my research in Project Rephetio applied a meta<ath-based approach to predict edges in a biomedical hetnet named Hetionet. 
<<<

→ We updated the introduction section to make it clearer that it primarily refers to recent embedding-based approaches (and hence cannot fully cover the wide variety of existing methods). 
We added mention of the metapath approach as well as the recently released scalable, rule-based AnyBURL system to the “Discussion and future work” section, as we plan to evaluate these methods as one of the next steps.

**Comment by dhimmel** [comment]
# Preprint v2 Review

Below is my review for [version 2 of the preprint](https://arxiv.org/abs/1912.04616v2). I just submitted this review to _Bioinformatics_.

***

Thank you for addressing the comments in my first review at https://github.com/OpenBioLink/OpenBioLink/issues/3. I've gone over my original feedback and confirmed the updates address my concerns.

The resource has grown substantially more interoperable and well documented since the initial version. I think it is now at a stage where the community might find it useful, and the application note is sufficient for publication.

Two small remaining items:

1. update the supplementary data PDF on GitHub with the revised supplement for v2 on arXiv (https://arxiv.org/abs/1912.04616v2).

2. when referencing Hetionet, the study cites https://doi.org/10.1371/journal.pcbi.1004259. However, the study that describes Hetionet is https://doi.org/10.7554/eLife.26726. Note that either study is a sufficient citation for metapath-based edge prediction, but only the eLife study describes Hetionet.
